jenkins-jobs/jenkins-backup/job-tpl.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project>
+  <actions/>
+  <description>Backs up Jenkins to S3</description>
+  <keepDependencies>false</keepDependencies>
+  <properties>
+    <jenkins.model.BuildDiscarderProperty>
+      <strategy class="hudson.tasks.LogRotator">
+        <daysToKeep>7</daysToKeep>
+        <numToKeep>7</numToKeep>
+        <artifactDaysToKeep>-1</artifactDaysToKeep>
+        <artifactNumToKeep>-1</artifactNumToKeep>
+      </strategy>
+    </jenkins.model.BuildDiscarderProperty>
+  </properties>
+  <scm class="hudson.scm.NullSCM"/>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers>
+    <hudson.triggers.TimerTrigger>
+      <spec>H 0 * * *</spec>
+    </hudson.triggers.TimerTrigger>
+  </triggers>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command># Delete all files in the workspace
+rm -rf *
+
+# Create a directory for the job definitions
+mkdir -p $BUILD_ID/jobs
+
+# Copy global configuration files into the workspace
+cp $JENKINS_HOME/*.xml $BUILD_ID/
+
+# Copy keys and secrets into the workspace
+cp $JENKINS_HOME/identity.key.enc $BUILD_ID/
+cp $JENKINS_HOME/secret.key $BUILD_ID/
+cp $JENKINS_HOME/secret.key.not-so-secret $BUILD_ID/
+cp -r $JENKINS_HOME/secrets $BUILD_ID/
+
+# Copy user configuration files into the workspace
+cp -r $JENKINS_HOME/users $BUILD_ID/
+
+# Copy custom Pipeline workflow libraries
+cp -r $JENKINS_HOME/workflow-libs $BUILD_ID
+
+# Copy job definitions into the workspace
+rsync -am --include=&apos;config.xml&apos; --include=&apos;*/&apos; --prune-empty-dirs --exclude=&apos;*&apos; $JENKINS_HOME/jobs/ $BUILD_ID/jobs/
+
+# Create an archive from all copied files (since the S3 plugin cannot copy folders recursively)
+tar czf jenkins-configuration.tar.gz $BUILD_ID/
+
+# Remove the directory so only the tar.gz gets copied to S3
+rm -rf $BUILD_ID
+
+export AWS_DEFAULT_REGION=&quot;${region}&quot;
+
+# Upload archive to S3
+echo &quot;Uploading archive to S3&quot;
+aws s3 cp jenkins-configuration.tar.gz s3://${jenkins_backup_s3_bucket}/${jenkins_backup_s3_key}/
+
+# Remove tar.gz after it gets uploaded to S3
+rm -rf jenkins-configuration.tar.gz</command>
+    </hudson.tasks.Shell>
+  </builders>
+  <publishers/>
+  <buildWrappers/>
+</project>
```

Comment:
Out of curiosity, did you look into any of the plugins/options in https://github.com/GSA/jenkins-deploy/issues/9? Not that it's _super_ complex, but would love to not have to maintain this ourselves.

Reply:
Yep, sure did. They didn’t really match my desired use case to store the backup in S3 without extra configuration anyway, so I didn’t see the point in maintaining a few extra components when a simple bash script could do it all ;)